teacherchannel/module.py

```diff
@@ -0,0 +1,258 @@
+            )
+            return
+        if channel.id != config.slave_id:
+            channel = ctx.guild.get_channel(config.slave_id)
+        await channel.set_permissions(teacher, read_messages=True, send_messages=True)
+        await ctx.reply(_(ctx, "Teacher added."))
+        await guild_log.info(
+            ctx.author,
+            ctx.channel,
+            f"Teacher {teacher.display_name} added to teacher channel {channel.name}.",
+        )
+
+    @commands.check(check.acl)
+    @teacherchannel_teacher.command(name="remove", aliases=["unset", "delete"])
+    async def teacherchannel_teacher_remove(
+        self, ctx, channel: nextcord.TextChannel, teacher: nextcord.Member
+    ):
+        """Unmark user as a teacher of a channel. Permission overrides will be removed from the teacher channel (slave).
+        Args:
+            channel: Either slave or master channel, both have the same effect.
+            teacher: Member which is no longer supposed to be marked as a teacher."""
+        success = TeacherChannelDB.remove_teacher(ctx.guild.id, channel.id, teacher.id)
+        if not success:
+            await ctx.reply(
+                _(
+                    ctx,
+                    "Could not remove teacher that was not there in the first place.",
+                )
+            )
+            return
+        config = TeacherChannelDB.get(ctx.guild.id, channel.id)
+        if channel.id != config.slave_id:
+            channel = ctx.guild.get_channel(config.slave_id)
+        await channel.set_permissions(teacher, overwrite=None)
+        await ctx.reply(_(ctx, "Teacher removed."))
+        await guild_log.info(
+            ctx.author,
+            ctx.channel,
+            f"Teacher {teacher.display_name} removed from teacher channel {channel.name}.",
+        )
+
+    @tasks.loop(seconds=10.0, count=1)
+    async def load_deltas(self):
+        guild_ids = [t.guild_id for t in TeacherChannelDB.get_guild_ids()]
+        for guild_id in guild_ids:
+            guild = self.bot.get_guild(guild_id)
+            channels_to_sync = TeacherChannelDB.get_all(guild_id)
+            for teacherchannel in channels_to_sync:
+                master = guild.get_channel(teacherchannel.master_id)
+                slave = guild.get_channel(teacherchannel.slave_id)
+                if not master:
+                    if slave is not None:
+                        name_to_log = slave.name
+                        await slave.delete()
+                    else:
+                        name_to_log = "<#" + str(teacherchannel.slave_id) + ">"
+                    await guild_log.warning(
+                        None,
+                        None,
+                        f"Teacher channel {name_to_log} cannot be synced, channel #<{teacherchannel.master_id}> "
+                        f"does not exist! Deleting from db.",
+                    )
+                    TeacherChannelDB.remove_channel(guild_id, teacherchannel.master_id)
```

Comment:
You already have `TeacherChannelDB` instance in `teacherchannel` variable. I think that you should add function `delete(self)` into `TeacherChannelDB` using `session.delete(self)` and calling `teacherchannel.delete()` here.